flink-test-utils/src/main/java/org/apache/flink/test/util/TestBaseUtils.java

```diff
@@ -310,6 +310,20 @@ public static void compareResultsByLinesInMemory(String expectedResultStr, Strin
 		Assert.assertArrayEquals(expected, result);
 	}
 
+	public static void compareResults(List<?> results, String expectedResults) {
```

Comment:
Isn't `compareResultAsText(List<T> result, String expected)` what you need? I don't think you need a new method in `TestBaseUtils`.